bluemira/equilibria/harmonics.py

```diff
@@ -0,0 +1,682 @@
+        )
+        return 1
+
+    # If the two LCFSs have identical coordinates then return a perfect fit metric
+    if np.array_equal(coords1.x, coords2.x) and np.array_equal(coords1.z, coords2.z):
+        bluemira_print("Perfect match! Original LCFS = SH approx LCFS")
+        return 0
+
+    # Get area of within the original and the SH approx LCFS
+    area1 = get_area_2d(coords1.x, coords1.z)
+    area2 = get_area_2d(coords2.x, coords2.z)
+
+    # Find intersections of the LCFSs
+    xcross, zcross = get_intersect(coords1.xz, coords2.xz)
+
+    # Check there are an even number of intersections
+    if np.mod(len(xcross), 2) != 0:
+        bluemira_print(
+            "Odd number of intersections for input and SH approx LCFS: this shouldn''t be possible. Trying again with more degrees."
+        )
+        return 1
+
+    # If there are no intersections then...
+    if len(xcross) == 0:
+        # Check if one LCFS is entirely within another
+        test_1_in_2 = polygon_in_polygon(coords2.xz.T, coords1.xz.T)
+        test_2_in_1 = polygon_in_polygon(coords1.xz.T, coords2.xz.T)
+        if all(test_1_in_2) or all(test_2_in_1):
+            # Calculate the metric if one is inside the other
+            return (np.max([area1, area2]) - np.min([area1, area2])) / (area1 + area2)
+        else:
+            # Otherwise they are in entirely different places
+            bluemira_print(
+                "The approximate LCFS does not overlap with the original. Trying again with more degrees."
+            )
+            return 1
+
+    # Calculate the area between the intersections of the two LCFSs,
+    # i.e., area within one but not both LCFSs.
+    c1 = Coordinates({"x": coords1.x, "z": coords1.z})
+    c2 = Coordinates({"x": coords2.x, "z": coords2.z})
+    c1_face = BluemiraFace(make_polygon(c1, closed=True))
+    c2_face = BluemiraFace(make_polygon(c2, closed=True))
+    result1 = boolean_cut(c1_face, c2_face)
+    result2 = boolean_cut(c2_face, c1_face)
+
+    #  Calculate metric
+    return (sum([f.area for f in result1]) + sum([f.area for f in result2])) / (
+        c1_face.area + c2_face.area
+    )
+
+
+def coils_outside_sphere_vacuum_psi(
+    eq: Equilibrium,
+) -> Tuple[np.ndarray, np.ndarray, CoilSet]:
+    """
+    Calculate the poloidal flux (psi) contribution from the vacuumn/coils
+    located outside of the sphere containing the plamsa, i.e., LCFS of
+    equlibrium state. N.B., currents from coilset are not considered here.
+
+    Parameters
+    ----------
+    eq: Bluemira Equilibrium
+        Starting equilibrium to use for our approximation
+
+    Returns
+    -------
+    vacuum_psi: ndarray
+        Psi contributuion from control coils
+        (only control coils! - be careful how you use it)
+    plasma_psi: ndarray
+        Psi contribution from plasma
+    new_coilset: Bluemira CoilSet
+        Coilset with control coils selected appropriately for use of SH approximation
```

Comment:
For methods with multiple returns:
```suggestion
    eq:
        Starting equilibrium to use for our approximation

    Returns
    -------
    vacuum_psi:
        Psi contributuion from control coils
        (only control coils! - be careful how you use it)
    plasma_psi:
        Psi contribution from plasma
    new_coilset:
        Coilset with control coils selected appropriately for use of SH approximation
```